src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/AWSTracingPipelineHandler.cs

```diff
@@ -0,0 +1,270 @@
+        {
+            carrier[name] = value;
+        };
+
+        private static readonly Dictionary<string, string> ServiceParameterMap = new Dictionary<string, string>()
+        {
+            { "DynamoDBv2", "TableName" },
+            { "SQS", "QueueUrl" },
+        };
+
+        private static readonly Dictionary<string, string> ParameterAttributeMap = new Dictionary<string, string>()
+        {
+            { "TableName", AWSSemanticConventions.AttributeAWSDynamoTableName },
+            { "QueueUrl", AWSSemanticConventions.AttributeAWSSQSQueueUrl },
+        };
+
+        private static readonly ActivitySource AWSSDKActivitySource = new ActivitySource(ActivitySourceName);
+
+        private readonly AWSClientInstrumentationOptions options;
+
+        public AWSTracingPipelineHandler(AWSClientInstrumentationOptions options)
+        {
+            this.options = options;
+        }
+
+        public override void InvokeSync(IExecutionContext executionContext)
+        {
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                base.InvokeSync(executionContext);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+        }
+
+        public override async Task<T> InvokeAsync<T>(IExecutionContext executionContext)
+        {
+            T ret = null;
+
+            var activity = this.ProcessBeginRequest(executionContext);
+            try
+            {
+                ret = await base.InvokeAsync<T>(executionContext).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                if (activity != null)
+                {
+                    this.ProcessException(activity, ex);
+                }
+
+                throw;
+            }
+            finally
+            {
+                if (activity != null)
+                {
+                    this.ProcessEndRequest(executionContext, activity);
+                }
+            }
+
+            return ret;
+        }
+
+        private Activity ProcessBeginRequest(IExecutionContext executionContext)
+        {
+            Activity activity = null;
+
+            var requestContext = executionContext.RequestContext;
+            var service = this.GetAWSServiceName(requestContext);
+            var operation = this.GetAWSOperationName(requestContext);
+
+            activity = AWSSDKActivitySource.StartActivity(service + "." + operation, ActivityKind.Client);
+
+            if (activity == null)
+            {
+                return null;
```

Comment:
You'd want to still propagate the context from Activity.Current

Reply:
When and why would we still want to propagate the context if the instrumentation activity is not created?

Reply:
Even if this activity is null (due to sampling), there could be a parent activity whose context should be propagated.

See example:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/MicroserviceExample/Utils/Messaging/MessageSender.cs#L62

Reply:
In a X-Ray service map, the spans created by downstream AWS managed services will be linked to this span as its parent. If this is not sampled and we still propagate the context to the downstream service, it may lead to incorrect parenting logic in X-Ray.
I'll have to verify the behavior before making the change.